radixdlt-core/radixdlt/src/main/java/com/radixdlt/store/berkeley/BerkeleyLedgerEntryStore.java

```diff
@@ -732,12 +760,64 @@ private void executeStateUpdate(com.sleepycat.je.Transaction txn, REStateUpdate
 				var buf2 = stateUpdate.getStateBuf();
 				var value = new DatabaseEntry(buf2.array(), buf2.position(), buf2.remaining());
 				resourceDatabase.putNoOverwrite(txn, new DatabaseEntry(addr.getBytes()), value);
+			} else if (stateUpdate.getParsed() instanceof VirtualParent) {
+				var p = (VirtualParent) stateUpdate.getParsed();
+				var typeByte = p.getData()[0];
+				if (typeByte != SubstateTypeId.UNCLAIMED_READDR.id()) {
+					var mapKey = SystemMapKey.ofValidatorDataParent(typeByte);
+					var key = new DatabaseEntry(mapKey.array());
+					var value = new DatabaseEntry(stateUpdate.getId().asBytes());
+					var result = mapDatabase.putNoOverwrite(txn, key, value);
```

Comment:
Logic above repeated few times, perhaps it worth to extract it into dedicated method.

Reply:
good point